src/Microsoft.TemplateEngine.Cli/TemplateSearch/CliTemplateSearchCoordinator.cs

```diff
@@ -179,9 +189,21 @@ private static bool ValidateCommandInput(SearchCommandArgs commandArgs)
                 Reporter.Error.WriteLine(LocalizableStrings.CliTemplateSearchCoordinator_Error_NoTemplateName.Red().Bold());
                 Reporter.Error.WriteLine(string.Format(LocalizableStrings.CliTemplateSearchCoordinator_Info_SearchHelp, string.Join(", ", SearchCommand.SupportedFilters.Select(f => $"'{f.OptionFactory().Aliases.First()}'"))));
                 Reporter.Error.WriteLine(LocalizableStrings.Generic_ExamplesHeader);
-                Reporter.Error.WriteCommand(CommandExamples.SearchCommandExample(commandArgs.CommandName, usePlaceholder: true));
-                Reporter.Error.WriteCommand(CommandExamples.SearchCommandExample(commandArgs.CommandName, additionalArgs: new[] { "--author", "Microsoft" }));
-                Reporter.Error.WriteCommand(CommandExamples.SearchCommandExample(commandArgs.CommandName, usePlaceholder: true, additionalArgs: new[] { "--author", "Microsoft" }));
+                Reporter.Error.WriteCommand(
+                    Example
+                        .For<NewCommand>(commandArgs.ParseResult)
+                        .WithSubcommand<SearchCommand>()
+                        .WithArgument(SearchCommand.NameArgument)
```

Comment:
I'm wondering since this are examples like we give examples in `dotnet new`:
```
Examples:
    dotnet new mvc --auth Individual
    dotnet new webapi
    dotnet new --help
    dotnet new nunit --help
```
If it would make sense to give more concrete examples here like:
```
   dotnet-new3 new3 search web
   dotnet-new3 new3 search web --author Microsoft
```
instead of
```
   dotnet-new3 new3 search [<template-name>]
   dotnet-new3 new3 search [<template-name>] --author Microsoft
```

Reply:
Changed to:
```
   dotnet-new3 new3 search web
   dotnet-new3 new3 search --author Microsoft
   dotnet-new3 new3 search web --language C#
```